src/main/java/appeng/recipes/IAERecipeFactory.java

```diff
@@ -0,0 +1,13 @@
+
+package appeng.recipes;
+
+
+import com.google.gson.JsonObject;
+
+import net.minecraftforge.common.crafting.JsonContext;
+
+
+public interface IAERecipeFactory
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Annotate the "IAERecipeFactory" interface with the @FunctionalInterface annotation [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1609)